tt-train/sources/examples/mnist_mlp/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+project(mnist_mlp)
+
+set(SOURCES
+    main.cpp
+    utils.cpp
+    models.cpp
+)
+
+CPMAddPackage(NAME mnist_dataset GITHUB_REPOSITORY wichtounet/mnist GIT_TAG master)
+include_directories(${mnist_dataset_SOURCE_DIR}/include)
```

Comment:
Does `mnist_dataset` not have a target that you can depend on instead of fishing out magic directories?

... nope.  They went and wrote a nice little Config.cmake file ... and used variables instead of a nicely packaged target 🙈  Ripe for a contribution upstream 😄

Reply:
Every new dependency had its own issue which need to be fixed in a unique way. So I am not sure.